### PR TITLE
fix(oscal): ensure component definition UUID is updated on modification

### LINKED ISSUE #614: Updating UUID when artifact is modified
Per [oscal note to developers](https://pages.nist.gov/OSCAL/resources/concepts/layer/implementation/component-definition/#important-note-to-developers) we should be updating oscal artifacts when we make automated modifications to the artifacts. 

component merge during generation
assessment results merge

### PR DESCRIPTION
**Pull request by brandtkeller**
## Description

Reading OSCAL docs and noted that the component definition/assessment results UUID needs to update whenever the file is modified. 

Assessment results was already responding accordingly - component definition had a commented line for performing this update.

Changed some error checking on merge operations to not be dead code. 

## Related Issue

Fixes #614 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Other (security config, docs update, etc)

## Checklist before merging

- [x] Test, docs, adr added or updated as needed
- [x] [Schema Updates](https://github.com/defenseunicorns/lula/blob/main/docs/community-and-contribution/schema-updates.md) applied
- [x] [Contributor Guide Steps](https://github.com/defenseunicorns/lula/blob/main/CONTRIBUTING.md) followed

### CODE REVIEW

File: src/pkg/common/oscal/complete-schema.go [hunk 1]
```diff
@@ -111,26 +111,39 @@ func MergeOscalModels(existingModel *oscalTypes_1_1_2.OscalModels, newModel *osc
 
 	// Component definition
 	if modelType == "component" {
+
+		if existingModel.ComponentDefinition == nil && newModel.ComponentDefinition == nil {
```
Comment (meganwolf0) [inline src/pkg/common/oscal/complete-schema.go]: Is this redundant with what's in `MergeComponentDefinitions`?

File: src/pkg/common/oscal/complete-schema.go [hunk 2]
```diff
@@ -111,26 +111,39 @@ func MergeOscalModels(existingModel *oscalTypes_1_1_2.OscalModels, newModel *osc
...
 		// Re-assign after processing errors
 		existingModel.ComponentDefinition = merged
-	} else if existingModel.ComponentDefinition == nil && newModel.ComponentDefinition != nil {
-		existingModel.ComponentDefinition = newModel.ComponentDefinition
 	}
 
 	// Assessment Results
 	if modelType == "assessment-results" {
+
+		if existingModel.AssessmentResults == nil && newModel.AssessmentResults == nil {
```
Comment (meganwolf0) [inline src/pkg/common/oscal/complete-schema.go]: Similar train of thought to above but should this logic be in the `MergeAssessmentResults` function?

File: src/pkg/common/oscal/complete-schema.go (hunk 1 above)
Comment (brandtkeller) [reply src/pkg/common/oscal/complete-schema.go]: good call - removed

File: src/pkg/common/oscal/complete-schema.go (hunk 2 above)
Comment (brandtkeller) [reply src/pkg/common/oscal/complete-schema.go]: removed

**Review by meganwolf0 — APPROVED** [review]

**Pull request merged**